ballerina/http_client_endpoint.bal

```diff
@@ -88,14 +88,17 @@ public client isolated class Client {
     } external;
 
     private isolated function processPost(string path, RequestMessage message, TargetType targetType,
-            string? mediaType, map<string|string[]>? headers) returns Response|anydata|ClientError {
+            string? mediaType, map<string|string[]>? headers) returns Response|StatusCodeResponse|anydata|ClientError {
         Request req = check buildRequest(message, mediaType);
         populateOptions(req, mediaType, headers);
         Response|ClientError response = self.httpClient->post(path, req);
         if observabilityEnabled && response is Response {
             addObservabilityInformation(path, HTTP_POST, response.statusCode, self.url);
         }
-        return processResponse(response, targetType, self.requireValidation);
+        if response is ClientError {
+            return response;
+        }
+        return externProcessResponse(response, targetType, self.requireValidation);
```

Comment:
This code segment is repeated among all the methods. Can't we keep the `processResponse` function and include this logic into it ?

Reply:
Fixed via [bce164f](https://github.com/ballerina-platform/module-ballerina-http/pull/1915/commits/bce164f18e1e33b548e2db2aa20f294e437a2202)